feat(mouse handler): allow user to swap resize and move using ShiftMask

barkyq opened the pull request:
# Description

Make it so `shift + mousekey + (left) click + drag` resizes windows and `shift + mousekey + (right) click + drag` moves windows. 

Remove static quantities from state struct.

Fixes #1213 

## Type of change

- [X] Development change (no change visible to user)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] This change requires a documentation update

## Updated user documentation:

Holding down `mousekey` while left-clicking is used to move windows, and holding down `mousekey` while right-clicking is used to resize windows. Alternatively, one can hold `mousekey` and Shift while left-clicking to resize windows or `mousekey` and Shift while right-clicking to move windows.

# Checklist:

- [X] Ran `make test` locally with no errors or warnings reported
  Note: To fully reproduce CI checks, you will need to run `make test-full-nix`. Usually, this is not necessary.
- [ ] Manual page has been updated accordingly
- [ ] Wiki pages have been updated accordingly (to perform **after** merge)

barkyq commented:
Some of my changes are opinionated, and happy to change anything which rubs you the wrong way!

mautamu commented:
This looks good to me but will wait to merge to see if anyone else has any comments since I'm admittedly not a heavy user of this functionality. Thanks again for all your changes this week :) !

Feel free to rebase and force push [7d86f72](https://github.com/leftwm/leftwm/pull/1217/commits/7d86f727a27de7b08c8569ffa896216623b62973) away if you'd like.

Eskaan reviewed (changes requested):
You don't have to move the implementation to the Manager to get direct access to the config, we already load it into the State struct.

barkyq commented:
I guess I was confused why these configuration settings are loaded into the `state`, since these settings do not change. As far as I can tell, the state should be for storing information which is changed by the program. 

Moreover, most of the other methods in display_event_handler.rs use the `Manager` rather than the state.

barkyq commented:
related to the discussion in #1219 about what config variables should be stored in `State` struct.

NicTanghe commented:
is this dead?

mautamu committed:
feat(mouse handler): allow user to swap resize and move using ShiftMask

- rebase to main, correct modmask for Shift

mouse combo handler

remove static quantities from state struct

fix: clippies

Co-authored-by: Dylan Cant <dylan@dylancant.ca>
Co-authored-by: barkyq <122579762+barkyq@users.noreply.github.com>

mautamu commented:
@NicTanghe This is ready to test if you'd like. Should be working now.

mautamu commented on leftwm-core/src/utils/modmask_lookup.rs:
```diff
@@ -8,7 +8,7 @@ bitflags! {
         const Zero = 0;
         const Any = 1;
-        const Shift = 1 << 1;
+        const Shift = 1 << 0;
```
Saw this in https://tronche.com/gui/x/xlib/events/keyboard-pointer/keyboard-pointer.html

This is the part that probably needs reviewed.

mautamu committed:
feat: add Caps Lock MM
